apps/channels/tasks.py

```diff
@@ -27,3 +27,30 @@ def handle_whatsapp_message(message_data: str):
         return
     message_handler = WhatsappChannel(experiment_channel=experiment_channel)
     message_handler.new_user_message(message)
+
+
+@shared_task
+def handle_facebook_message(message_data: str):
+    data = json.loads(message_data)
+    message = data["entry"][0]["messaging"][0]
+    page_id = message["recipient"]["id"]
+    attachments = message["message"].get("attachments", [])
+    content_type = None
+    media_url = None
+    if len(attachments) > 0:
+        attachment = attachments[0]
+        media_url = attachment["payload"]["url"]
+        content_type = attachment["type"]
+
+    message = FacebookMessage(
+        user_id=message["sender"]["id"],
+        page_id=page_id,
+        message_text=message["message"].get("text", ""),
+        media_url=media_url,
+        content_type=content_type,
+    )
+    experiment_channel = ExperimentChannel.objects.filter(extra_data__contains={"page_id": message.page_id}).first()
```

Comment:
could this also use the `filter_extras` manager method?

Reply:
Ah thanks

Reply:
https://github.com/dimagi/open-chat-studio/pull/115/commits/ce2298bedab43b56ec9d9aeb6e4fb6a1a217534e